spring-integration-sftp/src/test/java/org/springframework/integration/sftp/session/SftpSessionFactoryTests.java

```diff
@@ -126,4 +131,25 @@ public void concurrentGetSessionDoesntCauseFailure() throws IOException {
 		}
 	}
 
+	@Test
+	void externallyProvidedSshClientShouldNotHaveItsConfigurationOverwritten() throws IOException {
+		try (SshServer server = SshServer.setUpDefaultServer()) {
+			server.setPasswordAuthenticator((arg0, arg1, arg2) -> true);
+			server.setPort(0);
+			server.setKeyPairProvider(new SimpleGeneratorHostKeyProvider(new File("hostkey.ser").toPath()));
+			server.setSubsystemFactories(Collections.singletonList(new SftpSubsystemFactory()));
+			server.start();
+
+			SshClient externalClient = SshClient.setUpDefaultClient();
+			externalClient.setServerKeyVerifier(AcceptAllServerKeyVerifier.INSTANCE);
+			externalClient.setPasswordIdentityProvider(PasswordIdentityProvider.wrapPasswords("pass"));
+
+			DefaultSftpSessionFactory sftpSessionFactory = new DefaultSftpSessionFactory(externalClient, false);
+			sftpSessionFactory.setHost("localhost");
+			sftpSessionFactory.setPort(server.getPort());
+			sftpSessionFactory.setUser("user");
+
+			assertDoesNotThrow(() -> sftpSessionFactory.getSession().connect());
```

Comment:
I'm not sure that this test really reflects what we would expect from its name.
I don't even think that we need to test against the real server and make connections.
There is probably just enough to verify against some mock that its method calls don't happen or so.

Reply:
I have thought about using some mock.
But the init code does start a connection and tries to authenticate.

What you would want is something like this:
```java
@Test
void externallyProvidedSshClientShouldNotHaveItsConfigurationOverwritten() throws IOException {
	SshClient externalClient = spy(SshClient.setUpDefaultClient());

	DefaultSftpSessionFactory sftpSessionFactory = new DefaultSftpSessionFactory(externalClient, false);
	sftpSessionFactory.setHost("localhost");
	sftpSessionFactory.setUser("user");

	assertThatNoException().isThrownBy(() -> sftpSessionFactory.getSession());

	verify(externalClient, never()).setServerKeyVerifier(any());
	verify(externalClient, never()).setPasswordIdentityProvider(any());
	verify(externalClient, never()).setKeyIdentityProvider(any());
	verify(externalClient, never()).setUserInteraction(any());
}
```

But that would raise an exception:
```stacktrace
java.lang.IllegalStateException: failed to create SFTP Session
	at org.springframework.integration.sftp.session.DefaultSftpSessionFactory.getSession(DefaultSftpSessionFactory.java:292)
...
Caused by: org.apache.sshd.common.SshException: No more authentication methods available
...
	at org.springframework.integration.sftp.session.DefaultSftpSessionFactory.initClientSession(DefaultSftpSessionFactory.java:319)
	at org.springframework.integration.sftp.session.DefaultSftpSessionFactory.getSession(DefaultSftpSessionFactory.java:282)
	... 89 more
Caused by: org.apache.sshd.common.SshException: No more authentication methods available
	at org.apache.sshd.client.session.ClientUserAuthService.tryNext(ClientUserAuthService.java:379)
```

I doubt it would be wise to mock the better part of the calls which happen in the init towards the `SshClient`.

Any suggestions?

Reply:
Yeah... I see: the integration with that `SshClient` API is very heavy, therefore it is probably better to exchange a performance of the test in favor of clear code and direct calls instead of our error-prone mocks.
So, never mind then and thank you for your thoughts!
Please, consider to fix an assert in favor of `assertThatNoException()` - and we are good to merge this.